Gems/AtomTressFX/Code/Builders/HairAssetBuilder.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Assets/HairAsset.h>
+#include <Builders/HairAssetBuilder.h>
+#include <AssetBuilderSDK/SerializationDependencies.h>
+
+#include <AzCore/IO/FileIO.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        namespace Hair
+        {
+            void HairAssetBuilder::RegisterBuilder()
+            {
+                // Setup the builder descriptor
+                AssetBuilderSDK::AssetBuilderDesc builderDesc;
+                builderDesc.m_name = "HairAssetBuilder";
+
+                builderDesc.m_patterns.emplace_back(AssetBuilderSDK::AssetBuilderPattern(
+                    AZStd::string::format("*.%s", AMD::TFXFileExtension), AssetBuilderSDK::AssetBuilderPattern::PatternType::Wildcard));
+                builderDesc.m_busId = azrtti_typeid<HairAssetBuilder>();
+                builderDesc.m_version = 3;
+                builderDesc.m_createJobFunction =
+                    AZStd::bind(&HairAssetBuilder::CreateJobs, this, AZStd::placeholders::_1, AZStd::placeholders::_2);
+                builderDesc.m_processJobFunction =
+                    AZStd::bind(&HairAssetBuilder::ProcessJob, this, AZStd::placeholders::_1, AZStd::placeholders::_2);
+
+                BusConnect(builderDesc.m_busId);
+
+                AssetBuilderSDK::AssetBuilderBus::Broadcast(
+                    &AssetBuilderSDK::AssetBuilderBusTraits::RegisterBuilderInformation, builderDesc);
+            }
+
+            void HairAssetBuilder::ShutDown()
+            {
+                m_isShuttingDown = true;
+            }
+
+            void HairAssetBuilder::CreateJobs(
+                const AssetBuilderSDK::CreateJobsRequest& request, AssetBuilderSDK::CreateJobsResponse& response)
+            {
+                if (m_isShuttingDown)
+                {
+                    response.m_result = AssetBuilderSDK::CreateJobsResultCode::ShuttingDown;
+                    return;
+                }
+
+                for (const AssetBuilderSDK::PlatformInfo& info : request.m_enabledPlatforms)
+                {
+                    AssetBuilderSDK::JobDescriptor descriptor;
+                    descriptor.m_jobKey = AMD::TFXFileExtension;
+                    descriptor.m_critical = false;
+                    descriptor.SetPlatformIdentifier(info.m_identifier.c_str());
+                    response.m_createJobOutputs.push_back(descriptor);
+                }
```

Comment:
Is hair applicable to all platforms, or can it be omitted on servers?  If it can be ommitted on servers, then its possible to check what tags are on the platform - eg, 
```
if (info.HasTag("Server") 
{ 
    continue;
}
```
Its not mandatory to change this, but its an example of opting out based onthe platform if you want to.

Reply:
it's a good point but since the answer might differ in the future I decided to leave as is.  Hair is a beast that in the future I would much rather not do locally but currently there is no future address to that hence don't want to add limiting assumptions hard to detect aftermath when failed.